crates/subspace-networking/src/shared.rs

```diff
@@ -0,0 +1,44 @@
+use event_listener_primitives::Bag;
+use futures::channel::{mpsc, oneshot};
+use libp2p::kad::record::Key;
+use libp2p::{Multiaddr, PeerId};
+use parking_lot::Mutex;
+use std::sync::atomic::AtomicUsize;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub(crate) enum Command {
+    // TODO: We might want to have more specific gets eventually
+    GetValue {
+        key: Key,
+        result_sender: oneshot::Sender<Option<Vec<u8>>>,
+    },
+}
+
+#[derive(Default, Debug)]
+pub(crate) struct Handlers {
+    pub(crate) new_listener: Bag<Arc<dyn Fn(&Multiaddr) + Send + Sync + 'static>, Multiaddr>,
+}
+
+#[derive(Debug)]
+pub(crate) struct Shared {
```

Comment:
Shared means a sharable node? Would be nice to have a more explicit name :)

Reply:
I will probably add a comment in upcoming PRs. This is just an internal shared data structure between `Node` struct (primary public API of the library with convenient methods) and `NodeRunner`, which is the thing that drives internal libp2p implementation (swarm) forward. Since those 2 are related and need to communicate they need a few shared data structures.